### PR TITLE
Share Panel: Library panels feature

### LINKED ISSUE #90495: Share Panel: Library panels feature
Implement library panel drawer in panels

[Figma design](https://www.figma.com/design/AjvrAorYpnfSoMpU5kwf39/Adoption-of-Sharing?node-id=4560-57981&m=dev)

### PR DESCRIPTION
**Pull request by lucychen-grafana**
**What is this feature?**

Redesign the share library modal to a drawer

**Why do we need this feature?**

Part of the sharing redesign for better user experience

**Who is this feature for?**

Customers

**Which issue(s) does this PR fix?**:

Fixes #https://github.com/grafana/grafana/issues/90495

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by juanicabanas** [comment]
As there's no extra action inside the Drawer once the library panel is created, I think we should close the Drawer, as it is right now with the Modal

File: public/app/features/library-panels/components/AddLibraryPanelModal/AddLibraryPanelModal.tsx [hunk 1]
```diff
@@ -91,15 +93,25 @@ export const AddLibraryPanelContents = ({
...
-        <Button onClick={onCreate} disabled={invalidInput}>
-          <Trans i18nKey="library-panel.add-modal.create">Create library panel</Trans>
-        </Button>
-      </Modal.ButtonRow>
+      {config.featureToggles.newDashboardSharingComponent ? (
+        <Stack gap={1} justifyContent={'start'}>
+          <Button onClick={onCreate} disabled={invalidInput}>
+            <Trans i18nKey="share-panel.new-library-panel.create-button">Create library panel</Trans>
+          </Button>
+          <Button variant="secondary" onClick={onCancelClick} fill="outline">
```
[the remark this answers is not in the document]
Comment (lucychen-grafana) [reply public/app/features/library-panels/components/AddLibraryPanelModal/AddLibraryPanelModal.tsx]: OnDismiss doesn't work with the new drawer

File: public/app/plugins/panel/logs/panelcfg.cue
Comment (juanicabanas) [inline public/app/plugins/panel/logs/panelcfg.cue:1]: I don't think this file should be modified

**Review by juanicabanas — APPROVED** [review]

**Pull request merged**